zig/private/common/zig_build.bzl

```
@@ -195,6 +201,10 @@ def zig_build_impl(ctx, *, kind):
         arguments = ["build-lib", args]
         mnemonic = "ZigBuildLib"
         progress_message = "Building %{input} as Zig library %{output}"
+    elif kind == "zig_shared_library":
+        arguments = ["build-lib", "-dynamic", args]
+        mnemonic = "ZigBuildLib"
+        progress_message = "Building %{input} as Zig shared library %{output}"
```

If I inspect the output of the test described above I see that the soname of the library doesn't match the actual output. This is likely to cause trouble when linking against this library, as the corresponding load commands will use the wrong name.
```
$ readelf -d bazel-bin/simple-shared-library/library.so
...
  Tag        Type                         Name/Value
 0x000000000000000e (SONAME)             Library soname: [libmain.so]
...
```

The Zig compiler accepts a flag to change the SONAME:
```
  -fsoname[=name]                Override the default SONAME value
```

The following change should do the right thing:
```suggestion
    elif kind == "zig_shared_library":
        arguments = ["build-lib", "-dynamic", "-fsoname" + dynamic.basename, args]
        mnemonic = "ZigBuildLib"
        progress_message = "Building %{input} as Zig shared library %{output}"
```